src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts

```diff
@@ -0,0 +1,132 @@
+    // that the title of the index page (which is usually "Overview")
+    if (indexMatches.length && indexMatches[0].path == pathString) {
+      return [{ title: navNode.title, path: indexMatches[0].path }]
+    }
+    // If we don't have a child route that serves as an index page,
+    // then we have an index-less category. We check whether
+    // the current path string matches the inferred path
+    // to the index-less category.
+    if (indexMatches.length == 0) {
+      const routesWithPaths = navNode.routes.filter((r) => {
+        return Boolean(r.path)
+      })
+      if (routesWithPaths.length) {
+        const inferredPathParts = routesWithPaths[0].path.split('/')
+        inferredPathParts.pop()
+        const inferredPath = inferredPathParts.join('/')
+        if (inferredPath == pathString) {
+          return [{ title: navNode.title }]
+        }
+      }
+    }
+    // Otherwise, continue searching in all child routes
+    return findAllPathMatchedNodes(navNode.routes, pathString, depth + 1)
+  }
+  // If it's a node with a path value,
+  // return the node if the path is a match
+  if (typeof navNode.path == 'string') {
+    return navNode.path === pathString ? [{ ...navNode }] : []
+  }
+  // Otherwise, it's a divider or a direct link,
+  // so return an empty array (no match)
+  return []
+}
+
+function findAllPathMatchedNodes(navNodes, pathString, depth = 0) {
+  return navNodes
+    .slice()
+    .map((node) => findPathMatchedNodes(node, pathString, depth))
+    .reduce((matches, acc) => acc.concat(matches), [])
+}
+
+function getPathMatchedNode(navNodes, pathString) {
+  const matches = findAllPathMatchedNodes(navNodes, pathString)
+  // If we have exactly one match, this is great, and expected
+  if (matches.length == 1) return matches[0]
+  // If we do not have exactly one match, we likely
+  // have a problem with the navData that was not caught
+  // by our docs-sidenav validation functions, and we should address it.
+  // ...
+  // If we have zero matches, we can't recover from this,
+  // so we should throw an error and break the build.
+  if (matches.length == 0) {
+    throw new Error(
+      `Missing breadcrumb path: Found zero matches for "${pathString}". Please ensure there is a node (or index-less category) with the path "${pathString}" in the provided navData.`
+    )
+  }
+  // If we find more than one node matched for a path,
+  // we can return the first match to still be
+  // able to render the breadcrumb bar.
+  // We take this fallback approach in production contexts,
+  // but throw an error in other envs to flag the navData issue.
+  if (process.env.HASHI_ENV === 'production') {
+    return matches[0]
+  } else {
+    throw new Error(
+      `Ambiguous breadcrumb path: Found ${matches.length} matches for "${pathString}". Please ensure there is exactly one node or index-less category with the path "${pathString}" in the provided navData.`
+    )
+  }
```

Comment:
This _kind of_ feels like it doesn't belong in the breadcrumbs logic, but I don't feel too strongly about that 💭

Reply:
+1 I'm on the fence here as well. As I was writing it, I was kind of like.... I feel like this should be guaranteed by the incoming `navData`. And feels duplicative of the validation checks we already have; if those are in place there should be no possibility of ambiguous or missing matches.

Rather than throw errors here, we could silently just resolve these ambiguous matches. Or I could just make this a simpler error throw, eg if `matches.length != 1` then throw a more generic error about invalid navData. Let me know what you think!

Reply:
Maybe dev-warnings would be appropriate here, and we can work to ensure that our nav data validation is up-to-par? With the eventual goal being these code paths _should never happen_.

Side note, it'd be really nice to have some validation that runs on PRs which comments back to the PR with any errors if the nav data validation fails. I feel like that'd be a really nice feedback loop for contributors, as opposed to parsing through Vercel build logs (which many don't always do).

Reply:
I dig the idea of dev-warnings here, and really dig the validation on PRs at the source (content repos) that flag any issues with `navData` 💯 

To clarify re: `dev-warning` - what do you think the best way to detect dev environments would be?

I'm thinking we may be able to rely on [Vercel's System Environment variables](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables), specifically `VERCEL_ENV`.... Though this would require us to remove the currently-hard-coded `VERCEL_ENV="production"` from our `.env` file in this project. Have taken that approach for now, so I've got:

```js
const IS_DEV = process.env.VERCEL_ENV !== 'production'

// If navData has ambiguous matches, warn in development.
  // We can fallback to returning the first match, and this should
  // be fine from a visitor perspective. Less urgent to fix these types of issues.
  if (matches.length > 1) {
    if (IS_DEV) {
      console.warn(
        `Ambiguous breadcrumb path: Found ${matches.length} matches for "${pathString}". Please ensure there is exactly one node or index-less category with the path "${pathString}" in the provided navData.`
      )
    }
    return matches[0]
  }
  // Otherwise, we have zero matches, which would mean a breadcrumb with missing parts.
  // This would feel broken to a site visitor, so should be a blocking
  // issue we need to fix, so we throw an error.
  throw new Error(
    `Missing breadcrumb path: Found zero matches for "${pathString}". Please ensure there is a node (or index-less category) with the path "${pathString}" in the provided navData.`
  )
```

Let me know if that looks good... I guess another option would be using `HASHI_ENV` instead of `VERCEL_ENV`, but unsure on why those both exist 🤔

Reply:
I think in this case we could just do `NODE_ENV !== 'production'`. Agreed we should clarify and/or cleanup `HASHI_ENV`/`VERCEL_ENV`